### DISCUSSION TITLE
Problem when deploying vue3 project on Github Pages (works 100% locally and partially on Github pages)

### DISCUSSION THREAD

**Discussion by Kebrouh** [community/community#60257] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi !

So I've code myself a website using vue3, i have a total of 6 different pages with components. After a lot of trouble i've finally managed to deploy it with a custom domain name. However, out of my 4 "category" pages, 2 of them seem to be missing, 

![image](https://github.com/community/community/assets/135240267/d6ae66e2-509a-48f7-9475-fe57fb05ad59)
(img of the console above, hope it works)

These 4 "category" pages are all very similar, and all work localy (so when i npm run dev and after i npm run build and live server the dist) hence the reason of my confusion for only 2 of them breaking.

i am kinda blocked by this problem, it is always the same 2 pages, I've check the github pages restriction and i dont seem to be breaking any of them (my "database" is internal so its not fetching an external API, so i dont beleive this is the problem)

Please help :)

**Comment by GhostOf0days** [comment]
If the website is dynamic, it will fail on Github Pages. Please don’t use Github Pages. Instead, use Netlift, Heroku, or Vercel. Use Netlify for simple use cases/websites. Use Vercel for complex use cases/websites.

**Reply by Kebrouh** [reply]
I thought dynamic was if the website needed an external API, I thought i was fine with an internal json, all 4 pages requires the use of the internal json and 2 of them are succesfully working. 
Its my first time deploying website so thats why i prefered staying with github for it since its all I'm familiar with

**Reply by GhostOf0days** [reply]
Regardless, Github Pages is terrible. Netlify is not hard to deploy with. All you have to do is import your project. Here’s a [guide](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/).

**Reply by Kebrouh** [reply]
I will give it a try ty

**Reply by GhostOf0days** [reply]
Also, for Netlify, make sure you have a _redirects file that says ```/* /index.html 200```.

**Reply by Kebrouh** [reply]
Well i gave Netlify a quick try and I'm getting the same error as when I'm trying to use a workflow with github pages action (where i ended up staying with from a root) where all my node modules are considered undefined

**Comment by yoannchaudet** [comment]
@Kebrouh I don't see a custom workflow in your repository. GitHub Pages by default only builds automatically Jekyll sites, not Vue3.

You will want to go with a custom workflow (i.e. to tell a workflow to run your build steps prior to doing the deployment). Here are some rough steps:

- Go to the settings pages for your repo, then the Pages section
- Change "Source" from "Branch" to "GitHub Actions"
- The user interface should suggest you some workflows for your site. Since we don't support Vue3 out of the box, you will want to customize that yourself. You could start with a next/nuxt workflow (the ones we support are all available here: https://github.com/actions/starter-workflows/tree/main/pages)

**Reply by Kebrouh** [reply]
I've already tried to do it with a workflow. It seemed very more advantageous since its fast and relies on a single push of the main but sadly the deployment would always fail 
![image](https://github.com/community/community/assets/135240267/e23922b4-25f0-422b-b129-3a925b84081a)
I beleive it had something to do with my node modules, for example bootstrap would be considered undefined and that would break the process
I've tried about 3 existing workflow and then tried my own with many variation but the result was alwyas the same

**Reply by yoannchaudet** [reply]
Like I said, you want to use one of the workflows in https://github.com/actions/starter-workflows/tree/main/pages.

**Reply by Kebrouh** [reply]
I've tried once again a few of the one on this list and the error seems to always be the same thing that on the build the process gets cancelled when dealing with my node modules, i've tried to bring some changes to the .yml but nothing changes

**Reply by yoannchaudet** [reply]
There you go 🇫🇷 friend: https://github.com/Kebrouh/pemana/compare/main...yoannchaudet:pemana:main.

You could use Linux hosted runners instead of the Windows ones (they are slower) but for that you will want to fix the casing in some of your file names. On Linux filenames are case sensitives.

**Reply by Kebrouh** [reply]
Your .yml worked ! thank you so much !!!

I was wondering if you knew how to make it so that the build wouldnt ignore the img i have that arent used directly in a page, it's img that are called throught a .json. I had the same problem when deploying my site manually and i where the npm run build would ignore the image not directly called from a page and i would have to manually put the img in the dist 
I'm trying to understand where the build goes so that is there isnt a way to include my missing image i can add them manually
